source/fab/steps/analyse.py

```diff
@@ -167,6 +177,21 @@ def run(self, artefact_store: Dict, config):
         analysed_files = self._parse_files(files=files)
         self._add_manual_results(analysed_files)
 
+        # shall we search the results for fortran programs and a c function called main?
+        if self.find_programs:
+            # find fortran programs
+            sets_of_programs = [af.program_defs for af in by_type(analysed_files, AnalysedFortran)]
+            self.root_symbols = list(chain(*sets_of_programs))
+
+            # find c main()
+            c_with_main = list(filter(lambda c: 'main' in c.symbol_defs, by_type(analysed_files, AnalysedC)))
+            if c_with_main:
+                self.root_symbols.append('main')
+                if len(c_with_main) > 1:
+                    raise FabException("multiple c main() functions found")
```

Comment:
It's okay for there to be multiple C programs defined by the source tree. If you don't want to handle that with this change then raise an issue or project card for it.

Reply:
:+1: made [a ticket ](https://github.com/orgs/metomi/projects/1/views/1?pane=issue&itemId=21047392)